File: Assets/ApertureCutsceneEditor/Components/Actions/Action_SetPosition.js
#pragma strict
@script AddComponentMenu("Cutscene/Actions/Set Position")

@script ExecuteInEditMode()

@CutsceneEventOverrideNameAttribute("Set Position")

class Action_SetPosition extends CutsceneAction {

	function SetPosition(position:Vector3) {
		if (transform.position != position)
			transform.position = position;
	}
	function SetPosition(transformPosition:Transform) {
		if (transform.position != transformPosition.position)
			transform.position = transformPosition.position;
	}
	
	//#if !UNITY_FLASH
	@CutsceneActionAttribute("Set Position X")
	//#endif
	function SetPositionX(value:float) {
		if (transform.position.x != value)
			transform.position.x = value;
	}
	
	//#if !UNITY_FLASH
	@CutsceneActionAttribute("Set Position Y")
	//#endif
	function SetPositionY(value:float) {
		if (transform.position.y != value)
			transform.position.y = value;
	}
	
	//#if !UNITY_FLASH
	@CutsceneActionAttribute("Set Position Z")
	//#endif
	function SetPositionZ(value:float) {
		if (transform.position.z != value)
			transform.position.z = value;
	}
}

File: CutsceneEditor/CutsceneEditor/CutsceneEditor/Components/Actions/Action_SetPosition.cs
using UnityEngine;

[AddComponentMenu("Cutscene/Actions/Set Position")]

[ExecuteInEditMode()]

[CutsceneEventOverrideNameAttribute("Set Position")]

class Action_SetPosition : CutsceneAction
{

    public void SetPosition(Vector3 position)
    {
        if (transform.position != position)
            transform.position = position;
    }
    public void SetPosition(Transform transformPosition)
    {
        if (transform.position != transformPosition.position)
            transform.position = transformPosition.position;
    }

    //#if !UNITY_FLASH
    [CutsceneActionAttribute("Set Position X")]
    //#endif
    public void SetPositionX(float value)
    {
        if (transform.position.x != value)
            transform.position = new Vector3(value, transform.position.y, transform.position.z);
    }

    //#if !UNITY_FLASH
    [CutsceneActionAttribute("Set Position Y")]
    //#endif
    public void SetPositionY(float value)
    {
        if (transform.position.y != value)
            transform.position = new Vector3(transform.position.x, value, transform.position.z);
    }

    //#if !UNITY_FLASH
    [CutsceneActionAttribute("Set Position Z")]
    //#endif
    public void SetPositionZ(float value)
    {
        if (transform.position.z != value)
            transform.position = new Vector3(transform.position.x, transform.position.y, value);
    }
}

